diff --git a/tools/c_rehash.in b/tools/c_rehash.in
index 4dd1b44..e151f45 100644
--- a/tools/c_rehash.in
+++ b/tools/c_rehash.in
@@ -11,11 +11,16 @@
 # Perl c_rehash script, scan all files in a directory
 # and add symbolic links to their hash values.
 
-my $dir = {- quotify1($config{openssldir}) -};
-my $prefix = {- quotify1($config{prefix}) -};
+use File::Basename;
+use File::Spec::Functions qw( canonpath );
+use Cwd 'abs_path';
+
+my $script_dir = abs_path(dirname(__FILE__));
+my $dir = abs_path("$script_dir/../ssl");
+my $prefix = abs_path("$script_dir/..");
 
 my $errorcount = 0;
-my $openssl = $ENV{OPENSSL} || "openssl";
+my $openssl = $ENV{OPENSSL} || "openssl.exe";
 my $pwd;
 my $x509hash = "-subject_hash";
 my $crlhash = "-hash";
