--- a/Makefile.in	2025-06-10 07:48:50.000000000 +0900
+++ b/Makefile.in	2026-04-05 14:41:30.880864700 +0900
@@ -34,10 +34,10 @@
 	cd lib && $(MAKE) all
 	cd srclib && $(MAKE) all
 	cd src && $(MAKE) all
-	cd po && $(MAKE) all
-	cd man && $(MAKE) all
-	if test -d tests; then cd tests && $(MAKE) all; fi
-	cd install-tests && $(MAKE) all
+# 	cd po && $(MAKE) all
+# 	cd man && $(MAKE) all
+# 	if test -d tests; then cd tests && $(MAKE) all; fi
+# 	cd install-tests && $(MAKE) all

 lib/localcharset.h :
 	builddir="`pwd`"; cd libcharset && $(MAKE) all && $(MAKE) install-lib libdir="$$builddir/lib" includedir="$$builddir/lib"
@@ -56,8 +56,8 @@
 	cd src && $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	if [ ! -d $(DESTDIR)$(includedir) ] ; then $(mkinstalldirs) $(DESTDIR)$(includedir) ; fi
 	$(INSTALL_DATA) include/iconv.h.inst $(DESTDIR)$(includedir)/iconv.h
-	cd po && $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
-	cd man && $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)' mandir='$(mandir)'
+# 	cd po && $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
+# 	cd man && $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)' mandir='$(mandir)'

 install-strip : lib/localcharset.h force
 	cd libcharset && $(MAKE) install-strip prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
@@ -66,8 +66,8 @@
 	cd src && $(MAKE) install-strip prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	if [ ! -d $(DESTDIR)$(includedir) ] ; then $(mkinstalldirs) $(DESTDIR)$(includedir) ; fi
 	$(INSTALL_DATA) include/iconv.h.inst $(DESTDIR)$(includedir)/iconv.h
-	cd po && $(MAKE) install-strip prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
-	cd man && $(MAKE) install-strip prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)' mandir='$(mandir)'
+# 	cd po && $(MAKE) install-strip prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
+# 	cd man && $(MAKE) install-strip prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)' mandir='$(mandir)'

 installdirs : force
 	cd libcharset && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
@@ -75,8 +75,8 @@
 	cd srclib && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	cd src && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	if [ ! -d $(DESTDIR)$(includedir) ] ; then $(mkinstalldirs) $(DESTDIR)$(includedir) ; fi
-	cd po && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
-	cd man && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)' mandir='$(mandir)'
+# 	cd po && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
+# 	cd man && $(MAKE) installdirs prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)' mandir='$(mandir)'

 uninstall : force
 	cd libcharset && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
@@ -84,17 +84,17 @@
 	cd srclib && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	cd src && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)'
 	$(RM) $(DESTDIR)$(includedir)/iconv.h
-	cd po && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
-	cd man && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)' mandir='$(mandir)'
+# 	cd po && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)'
+# 	cd man && $(MAKE) uninstall prefix='$(prefix)' exec_prefix='$(exec_prefix)' datarootdir='$(datarootdir)' datadir='$(datadir)' mandir='$(mandir)'

 check : all force
 	cd libcharset && $(MAKE) check
 	cd lib && $(MAKE) check
 	cd srclib && $(MAKE) check
 	cd src && $(MAKE) check
-	cd po && $(MAKE) check
-	cd man && $(MAKE) check
-	if test -d tests; then cd tests && $(MAKE) check; fi
+# 	cd po && $(MAKE) check
+# 	cd man && $(MAKE) check
+# 	if test -d tests; then cd tests && $(MAKE) check; fi

 mostlyclean : force
 	cd libcharset && $(MAKE) mostlyclean
@@ -102,10 +102,10 @@
 	cd lib && $(MAKE) mostlyclean
 	cd srclib && $(MAKE) mostlyclean
 	cd src && $(MAKE) mostlyclean
-	cd po && $(MAKE) mostlyclean
-	cd man && $(MAKE) mostlyclean
-	if test -d tests; then cd tests && $(MAKE) mostlyclean; fi
-	cd install-tests && $(MAKE) mostlyclean
+# 	cd po && $(MAKE) mostlyclean
+# 	cd man && $(MAKE) mostlyclean
+# 	if test -d tests; then cd tests && $(MAKE) mostlyclean; fi
+# 	cd install-tests && $(MAKE) mostlyclean
 	rm -rf bin

 clean : force
@@ -114,10 +114,10 @@
 	cd lib && $(MAKE) clean
 	cd srclib && $(MAKE) clean
 	cd src && $(MAKE) clean
-	cd po && $(MAKE) clean
-	cd man && $(MAKE) clean
-	if test -d tests; then cd tests && $(MAKE) clean; fi
-	cd install-tests && $(MAKE) clean
+# 	cd po && $(MAKE) clean
+# 	cd man && $(MAKE) clean
+# 	if test -d tests; then cd tests && $(MAKE) clean; fi
+# 	cd install-tests && $(MAKE) clean
 	rm -rf bin

 distclean : force
@@ -126,10 +126,10 @@
 	cd lib && if test -f Makefile; then $(MAKE) distclean; fi
 	cd srclib && if test -f Makefile; then $(MAKE) distclean; fi
 	cd src && if test -f Makefile; then $(MAKE) distclean; fi
-	cd po && if test -f Makefile; then $(MAKE) distclean; fi
-	cd man && if test -f Makefile; then $(MAKE) distclean; fi
-	if test -d tests; then cd tests && if test -f Makefile; then $(MAKE) distclean; fi; fi
-	cd install-tests && if test -f Makefile; then $(MAKE) distclean; fi
+# 	cd po && if test -f Makefile; then $(MAKE) distclean; fi
+# 	cd man && if test -f Makefile; then $(MAKE) distclean; fi
+# 	if test -d tests; then cd tests && if test -f Makefile; then $(MAKE) distclean; fi; fi
+# 	cd install-tests && if test -f Makefile; then $(MAKE) distclean; fi
 	$(RM) config.status config.log config.cache Makefile libtool
 	$(RM) config.h stamp-h1 lib/stamp-h2
 	$(RM) include/iconv.h include/iconv.h.inst
@@ -141,17 +141,17 @@
 	cd lib && if test -f Makefile; then $(MAKE) maintainer-clean; fi
 	cd srclib && if test -f Makefile; then $(MAKE) maintainer-clean; fi
 	cd src && if test -f Makefile; then $(MAKE) maintainer-clean; fi
-	cd po && if test -f Makefile; then $(MAKE) maintainer-clean; fi
-	cd man && if test -f Makefile; then $(MAKE) maintainer-clean; fi
-	if test -d tests; then cd tests && if test -f Makefile; then $(MAKE) maintainer-clean; fi; fi
-	cd install-tests && if test -f Makefile; then $(MAKE) maintainer-clean; fi
+# 	cd po && if test -f Makefile; then $(MAKE) maintainer-clean; fi
+# 	cd man && if test -f Makefile; then $(MAKE) maintainer-clean; fi
+# 	if test -d tests; then cd tests && if test -f Makefile; then $(MAKE) maintainer-clean; fi; fi
+# 	cd install-tests && if test -f Makefile; then $(MAKE) maintainer-clean; fi
 	$(RM) config.status config.log config.cache Makefile libtool
 	$(RM) config.h stamp-h1 lib/stamp-h2
 	$(RM) include/iconv.h include/iconv.h.inst
 	rm -rf bin

 installcheck : force
-	cd install-tests && $(MAKE) installcheck prefix='$(prefix)' includedir='$(includedir)' exec_prefix='$(exec_prefix)' bindir='$(bindir)' libdir='$(libdir)'
+# 	cd install-tests && $(MAKE) installcheck prefix='$(prefix)' includedir='$(includedir)' exec_prefix='$(exec_prefix)' bindir='$(bindir)' libdir='$(libdir)'


 # List of source files.
@@ -236,10 +236,10 @@
 	test -d '$(distdir)'/lib || mkdir '$(distdir)'/lib; cd lib && $(MAKE) distdir distdir='$(distdir)'/lib
 	test -d '$(distdir)'/srclib || mkdir '$(distdir)'/srclib; cd srclib && $(MAKE) distdir distdir='$(distdir)'/srclib
 	test -d '$(distdir)'/src || mkdir '$(distdir)'/src; cd src && $(MAKE) distdir distdir='$(distdir)'/src
-	test -d '$(distdir)'/tests || mkdir '$(distdir)'/tests; cd tests && $(MAKE) distdir distdir='$(distdir)'/tests
-	test -d '$(distdir)'/po || mkdir '$(distdir)'/po; cd po && $(MAKE) distdir distdir='$(distdir)'/po
-	test -d '$(distdir)'/man || mkdir '$(distdir)'/man; cd man && $(MAKE) distdir distdir='$(distdir)'/man
-	test -d '$(distdir)'/install-tests || mkdir '$(distdir)'/install-tests; cd install-tests && $(MAKE) distdir distdir='$(distdir)'/install-tests
+# 	test -d '$(distdir)'/tests || mkdir '$(distdir)'/tests; cd tests && $(MAKE) distdir distdir='$(distdir)'/tests
+# 	test -d '$(distdir)'/po || mkdir '$(distdir)'/po; cd po && $(MAKE) distdir distdir='$(distdir)'/po
+# 	test -d '$(distdir)'/man || mkdir '$(distdir)'/man; cd man && $(MAKE) distdir distdir='$(distdir)'/man
+# 	test -d '$(distdir)'/install-tests || mkdir '$(distdir)'/install-tests; cd install-tests && $(MAKE) distdir distdir='$(distdir)'/install-tests


 # Creating a distribution tarball.
--- a/src/Makefile.in	2025-07-07 18:26:22.000000000 +0900
+++ b/src/Makefile.in	2026-04-06 06:33:51.660428900 +0900
@@ -115,7 +115,7 @@
 	if [ ! -d $(DESTDIR)$(bindir) ] ; then $(mkinstalldirs) $(DESTDIR)$(bindir) ; fi
 	case "@host_os@" in \
 	  hpux*) $(CC) $(LDFLAGS) $(CFLAGS) $(iconv_LDFLAGS) iconv.@OBJEXT@ ../srclib/libicrt.a -L$(DESTDIR)$(libdir) -liconv @LIBINTL@ $(OBJECTS_RES_@WOE32@) `if test -n '$(DESTDIR)'; then echo " -Wl,+b -Wl,$(libdir)"; fi` -o iconv$(EXEEXT);; \
-	  *) $(LIBTOOL_LINK) $(CC) $(LDFLAGS) $(CFLAGS) $(iconv_LDFLAGS) iconv.@OBJEXT@ ../srclib/libicrt.a @LTLIBINTL@ $(DESTDIR)$(libdir)/libiconv.la $(OBJECTS_RES_@WOE32@) -o iconv$(EXEEXT);; \
+	  *) $(LIBTOOL_LINK) $(CC) $(CFLAGS) $(iconv_LDFLAGS) iconv.@OBJEXT@ ../srclib/libicrt.a @LTLIBINTL@ $(LDFLAGS) $(DESTDIR)$(libdir)/libiconv.la $(OBJECTS_RES_@WOE32@) -o iconv$(EXEEXT);; \
 	esac
 	$(INSTALL_PROGRAM_ENV) $(LIBTOOL_INSTALL) $(INSTALL_PROGRAM) iconv$(EXEEXT) $(DESTDIR)$(bindir)/iconv$(EXEEXT)

@@ -167,4 +167,3 @@
 	done

 force :
-
