diff --git a/meson.build b/meson.build
index 1ae14fc..5ccd38f 100644
--- a/meson.build
+++ b/meson.build
@@ -2246,7 +2246,7 @@ gvdb_dep = dependency('gvdb')
 libm = cc.find_library('m', required : false)
 libffi_dep = dependency('libffi', version : '>= 3.0.0')

-libz_dep = dependency('zlib')
+libz_dep = dependency('zlib', 'ZLIB', modules:['ZLIB::ZLIB'])

 libatomic_test_code = '''
   int main (int argc, char ** argv) {
