diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index cc800b3..8aff345 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -3,23 +3,6 @@ set(CMAKE_C_STANDARD 99)
 configure_file(config.h.in config.h)
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-if(NOT MSVC)
-  if(NOT WIN32)
-    find_library(MATH m)
-  else()
-    set(MATH "")
-  endif()
-  include(FindZLIB)
-else()
-  set(MATH "")
-  find_program(NUGET nuget)
-  if(NUGET)
-    execute_process(COMMAND ${NUGET} install zlib)
-  endif()
-  include_directories(
-    ${PROJECT_SOURCE_DIR}/windows/third-party/zlib-1.2.11/include/)
-endif()
-
 if(NOT MSVC)
   if(NOT WIN32)
     find_library(MATH m)
@@ -32,17 +15,9 @@ if(NOT MSVC)
   set(PKG_CONFIG_PRIVATELIBS "-lz ${PKG_CONFIG_PRIVATELIBS}")
 else()
   set(MATH "")
-  find_program(NUGET nuget)
-  if(NOT NUGET)
-    message(
-      FATAL
-      "Cannot find nuget command line tool.\nInstall it with e.g. choco install nuget.commandline"
-    )
-  else()
-    execute_process(COMMAND ${NUGET} install zlib)
-  endif()
-  include_directories(
-    ${PROJECT_SOURCE_DIR}/windows/third-party/zlib-1.2.11/include/)
+  include(FindZLIB)
+  include_directories(${ZLIB_INCLUDE_DIRS})
+  set(PKG_CONFIG_PRIVATELIBS "-lzlib ${PKG_CONFIG_PRIVATELIBS}")
 endif()
 
 set(libsrc
@@ -161,7 +136,7 @@ else()
                          mysofa_export.h)
 endif()
 
-# Must install export header so downstream packages can properly understand 
+# Must install export header so downstream packages can properly understand
 # which functions are accessible
 install(FILES ${PROJECT_BINARY_DIR}/src/mysofa_export.h
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
