--- a/CMakeLists.txt	2026-06-13 23:17:10.395211300 +0900
+++ a/CMakeLists.txt	2026-06-14 00:14:36.128725200 +0900
@@ -109,7 +109,7 @@
 check_include_file("windows.h" HAVE_WINDOWS_H)

 include(CheckLibraryExists)
-check_library_exists(z zlibVersion "" HAVE_LIBZ)
+check_library_exists(zlib zlibVersion "" HAVE_LIBZ)
 check_library_exists(lzo2 lzo1x_1_15_compress "" HAVE_LIBLZO2)
 check_library_exists(lz4 LZ4_compress_default "" HAVE_LIBLZ4)

@@ -305,7 +305,7 @@
   target_link_libraries(snappy_test_support snappy)

   if(HAVE_LIBZ)
-    target_link_libraries(snappy_test_support z)
+    target_link_libraries(snappy_test_support zlib)
   endif(HAVE_LIBZ)
   if(HAVE_LIBLZO2)
     target_link_libraries(snappy_test_support lzo2)
