diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1b96bf3..1885906 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,7 +96,7 @@ add_subdirectory(include)
 add_subdirectory(src)
 
 # pkg-config file
-if (UNIX OR MINGW)
+if (UNIX OR MINGW OR MSVC)
 configure_file(libssh.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/libssh.pc @ONLY)
 install(
   FILES
@@ -258,4 +258,3 @@ if (WITH_SERVER)
 message(STATUS "Global bind config: ${GLOBAL_BIND_CONFIG}")
 endif()
 message(STATUS "********************************************")
-
