--- a/include/gavl/gavldefs.h 2026-04-16 12:50:15.139721300 +0900
+++ b/include/gavl/gavldefs.h 2026-04-16 12:52:17.089720100 +0900
@@ -21,12 +21,19 @@
 #ifndef GAVLDEFS_H_INCLUDED
 #define GAVLDEFS_H_INCLUDED

-#ifdef DLLEXPORT
-#define GAVL_PUBLIC __declspec(dllexport)
+#if defined(_WIN32) || defined(__CYGWIN__)
+#  ifdef GAVL_BUILD_DLL
+#    define GAVL_PUBLIC __declspec(dllexport)
+#  else
+#    define GAVL_PUBLIC __declspec(dllimport)
+#  endif
 #else
-#define GAVL_PUBLIC __attribute__ ((visibility("default")))
-#endif
+#  if __GNUC__ >= 4
+#    define GAVL_PUBLIC __attribute__ ((visibility("default")))
+#  else
+#    define GAVL_PUBLIC
+#  endif
+#endif


 #endif // GAVLDEFS_H_INCLUDED
-
